src/puppetlabs/trapperkeeper/services/status/status_service.clj

```diff
@@ -0,0 +1,30 @@
+(ns puppetlabs.trapperkeeper.services.status.status-service
+  (:require [clojure.tools.logging :as log]
+            [compojure.core :as compojure]
+            [puppetlabs.trapperkeeper.core :refer [defservice]]
+            [puppetlabs.trapperkeeper.services :refer [service-context]]
+            [puppetlabs.trapperkeeper.services.status.status-core :as core]
+            [puppetlabs.trapperkeeper.services.webrouting.webrouting-service :as webrouting-service]))
+
+(defprotocol StatusService
+  (register-status [this service-name service-version status-version status-fn]
```

Comment:
> it might be useful to have helper functions that can pull out the service's version from the lein metadata.

Yeah, I think we'll probably want to do something like this eventually.

re: making `status-version` optional ...

I think we could come up with a reasonable default behavior for the case when no version is provided but multiple functions are registered.  Perhaps we just throw an error which tells the caller that, if you want to do that, you need to specify versions.  Or, we could just say that "last one wins", although I think throwing an error is probably better.  Either way, this issue falls into my bucket of "reasonable defaults" which is one of my favorite programming topics to rant about in code review.  :bicyclist: :smirk_cat: 

> this is actually also potentially a current problem - two status functions can be registered without changing the status version... 

I think this case should probably throw an error as well.  The only thing I can imagine that would cause this to happen would be a bug - I can't come up with any good reason for wanting to overwrite a previously-registered status function.